### PR TITLE
NXP-19912: Changes to run tests against an addon as backend

### PR DESCRIPTION
**Pull request by kevinleturc**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/nuxeo/nuxeo/pull/142" target="_blank">ReviewNinja

File: nuxeo-core/nuxeo-core-test/src/main/java/org/nuxeo/ecm/core/test/CoreFeature.java [hunk 1]
```diff
@@ -165,11 +163,14 @@ public void start(FeaturesRunner runner) {
...
             storageConfiguration.init();
+            for (String bundle : storageConfiguration.getExternalBundles()) {
+                harness.deployBundle(bundle);
+            }
             URL blobContribUrl = storageConfiguration.getBlobManagerContrib(runner);
             harness.getContext().deploy(new URLStreamRef(blobContribUrl));
             URL repoContribUrl = storageConfiguration.getRepositoryContrib(runner);
             harness.getContext().deploy(new URLStreamRef(repoContribUrl));
-        } catch (IOException e) {
+        } catch (Exception e) {
```
Comment (efge) [inline nuxeo-core/nuxeo-core-test/src/main/java/org/nuxeo/ecm/core/test/CoreFeature.java]: We don't like to catch `Exception`, so if you really have to (due to stupid `deployBundle` signature) do it right around the incriminated method.

File: nuxeo-core/nuxeo-core-test/src/main/java/org/nuxeo/ecm/core/test/StorageConfiguration.java [hunk 2]
```diff
@@ -177,6 +183,17 @@ protected void clearMongoDB(MongoDBRepositoryDescriptor descriptor) throws Unkno
...
     }
 
+    protected void initExternal() {
+        // Get DBSHelper by reflection
+        String className = String.format("org.nuxeo.ecm.storage.%s.DBSHelperImpl", coreType);
+        try {
+            dbsHelper = (DBSHelper) Class.forName(className).newInstance();
+            dbsHelper.init();
+        } catch (ReflectiveOperationException e) {
+            throw new ExceptionInInitializerError("DBSHelperImpl not found: " + className);
```
Comment (efge) [inline nuxeo-core/nuxeo-core-test/src/main/java/org/nuxeo/ecm/core/test/StorageConfiguration.java]: Please include the cause when throwing.

File: nuxeo-core/nuxeo-core-test/src/main/java/org/nuxeo/ecm/core/test/StorageConfiguration.java [hunk 3]
```diff
@@ -177,6 +183,17 @@ protected void clearMongoDB(MongoDBRepositoryDescriptor descriptor) throws Unkno
         }
     }
 
+    protected void initExternal() {
+        // Get DBSHelper by reflection
+        String className = String.format("org.nuxeo.ecm.storage.%s.DBSHelperImpl", coreType);
```
Comment (efge) [inline nuxeo-core/nuxeo-core-test/src/main/java/org/nuxeo/ecm/core/test/StorageConfiguration.java]: The current package for mongodb is org.nuxeo.ecm.core.storage.mongodb. Why not keep the "core" part?

Comment (kevinleturc) [inline nuxeo-core/nuxeo-core-test/src/main/java/org/nuxeo/ecm/core/test/StorageConfiguration.java]: We changed that when we extracted marklogic from Nuxeo repository, also nuxeo-core-storage-marklogic became nuxeo-storage-marklogic, does we get back on this ? (and all other changes to add core at right position)

**Pull request closed without merging**